bazel/src/main/java/com/google/api/codegen/bazel/BazelBuildFileView.java

```diff
@@ -137,8 +142,10 @@ class BazelBuildFileView {
                   .getOrDefault(bp.getProtoPackage() + "." + service, service)
               // Default service name as it appears in the proto.
               : service;
-      javaTests.add(javaPackage + "." + actualService + "ClientTest");
-      if ("grpc+rest".equals(transport)) {
+      if (javaTransport.contains("grpc")) {
+        javaTests.add(javaPackage + "." + actualService + "ClientTest");
+      }
+      if (javaTransport.contains("rest")) {
```

Comment:
Not directly related to this PR, due to the recent "no http annotation" issue, we may not generate any HttpJson tests if a service does not have any http eligible methods, so we may need to manually remove these tests after generation. Not sure if this tool can read protos or not, if not, we may have to live with it for now and see if there is any other workaround.

Reply:
This is a good idea! Would you mind filing a go/build-gen-feature ? This is definitely possible - we can check relatively easily via presence of imports for that annotation + presence of http rules in the service config.